### PR TITLE
Added mobile carrier info related semantic conventions to span spec

### PR DESCRIPTION
**Pull request by bryce-b**
Fixes #
Adds semantic conventions defining mobile carrier data defined in [#1647](https://github.com/open-telemetry/opentelemetry-specification/issues/1647)

## Changes
- adds `net.host.carrier.*`
- adds `net.host.connection_type`

Related issues #
#1647 : `Add semantic convention for describing network connectivity conditions`

### CODE REVIEW

File: specification/trace/semantic_conventions/span-general.md [hunk 1]
```diff
@@ -41,7 +41,25 @@ the `net.peer.*` properties of a client are equal to the `net.host.*` properties
 | `net.host.ip` | string | Like `net.peer.ip` but for the host IP. Useful in case of a multi-IP host. | `192.168.0.1` | No |
 | `net.host.port` | int | Like `net.peer.port` but for the host port. | `35555` | No |
 | `net.host.name` | string | Local hostname or similar, see note below. | `localhost` | No |
-
+| `net.host.connection_type` | string | The current network connection type in common terms | `wifi` | No |
+| `net.host.carrier.name` | string | Name of the mobile carrier. | `sprint` | No |
+| `net.host.carrier.mcc` | string | [Mobile country code](https://en.wikipedia.org/wiki/Mobile_country_code) | `310` | No | 
+| `net.host.carrier.mnc` | string | [Mobile network code](https://en.wikipedia.org/wiki/Mobile_country_code) | `001` | No |
+| `net.host.carrier.icc` | string | [ISO country code (two character)](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) | `DE` | No |
```
Comment (iNikem) [inline specification/trace/semantic_conventions/span-general.md:48]: These are not "general network connection attributes", but are very specific to mobile clients.

Comment (Oberon00) [reply specification/trace/semantic_conventions/span-general.md:48]: I think the "general" in the title is more about "low-level networking" as opposed to "protocol-specific attributes".

In any case, I think if we make them span attributes (which seems to be the best currently available option) then putting them under `net.host` is sensible.

Also `net.host.connection_type` would make sense for other clients as well.

File: specification/trace/semantic_conventions/span-general.md [hunk 2]
```diff
@@ -41,7 +41,25 @@ the `net.peer.*` properties of a client are equal to the `net.host.*` properties
...
+
+`net.host.connection_type` SHOULD be one of the following : 
+
+| Value | Description |
+| --- | --- |
+| `wifi` | connected via wifi |
```
Comment (Oberon00) [inline specification/trace/semantic_conventions/span-general.md:54]: What about adding a `wired` value?

Comment (Oberon00) [inline specification/trace/semantic_conventions/span-general.md:54]: How about also adding a `wired` value?

**Review by Oberon00 — COMMENTED** [review]
Semantic conventions are mostly edited in the YAML sources from which the markdown is then generated. See https://github.com/open-telemetry/opentelemetry-specification/tree/main/semantic_conventions

File: specification/trace/semantic_conventions/span-general.md (hunk 2 above)
Comment (bryce-b) [reply specification/trace/semantic_conventions/span-general.md:54]: sure!

**Review by nachoBonafonte — APPROVED** [review]

**Comment by carlosalberto** [comment]
Ping @bryce-b

File: specification/trace/semantic_conventions/span-general.md (hunk 1 above)
Comment (bryce-b) [reply specification/trace/semantic_conventions/span-general.md:48]: @iNikem do you think these should be placed in a different semantic convention document, specifically for mobile?

**Comment by jmacd** [comment]
Please re-open if you would like to resume this work.

**Pull request closed without merging**